### PR TITLE
split go-uaa

### PR DESCRIPTION
**Pull request by joefitzgerald**
Per the instructions in https://github.com/cloudfoundry/community/pull/801#issuecomment-2042393959, this splits go-uaa from the rest of the UAA repos so that I can continue to merge PRs as the original author and maintainer of the library.

/cc @strehle

### CODE REVIEW

**Review by strehle — APPROVED** [review]
@Tallicia  someone from your company should check because you use it more than we

**Comment by strehle** [comment]
@joefitzgerald if you have PR, please create them

**Comment by joefitzgerald** [comment]
@strehle please review the context from https://github.com/cloudfoundry/community/pull/801#issuecomment-2040111759. In summary:

* I created the go-uaa library
* I maintained it, and worked with the UAA team to integrate it into the UAA CLI while I was at Pivotal
* The UAA team has turned over a few times since then, but I have remained a maintainer of the library; I still aspire to have the UAA maintain the library but I remain a backstop for merging of PRs that are left open for too long
* The repo was moved without consultation from the cloudfoundry-community org (where I was an org owner) to the cloudfoundry org (where I did not have any commit rights), resulting in me losing the ability to maintain the library
* This PR restores my ability to maintain the library

**Comment by beyhan** [comment]
Thanks @joefitzgerald for the pr. I will take care of this.

**Review by rkoster — APPROVED** [review]

**Review by stephanme — APPROVED** [review]

**Review by ameowlia — APPROVED** [review]

**Review by ChrisMcGowan — APPROVED** [review]

**Comment by joefitzgerald** [comment]
The machine appears to be once again attempting to prevent me from being a backstop maintainer of the library I contributed to a different org. I do not have the access to merge pull requests in the `go-uaa` repo.  And I have been put on [a list of people to be removed](https://github.com/cloudfoundry/community/pull/1271) altogether from the approvers list.

1. why / how did I lose the ability to merge PRs in `go-uaa`
2. How can I be exempted from the annual requirement to make the same argument. It doesn’t seem like the community is growing, please don’t make it any harder than it needs to be for me to provide a service to the community.

**Comment by beyhan** [comment]
> 
> > We introduced branch protection with #1218 and most probably this is the reason you can't merge.
> 
> So how do I regain the ability to merge PRs in the `go-uaa` library?

Hi @joefitzgerald,

> I have been merging the occasional PR, but that doesn't count as "activity". Can you please update the logic to count that as "activity"?

What counts to an activity is defined by Github and documented on [this page](https://docs.github.com/en/account-and-profile/reference/why-are-my-contributions-not-showing-up-on-my-profile). We just use Github APIs for that. We don't have our special logic here. The proposal by @strehle sounds like it will add activity to your profile. 

The branch protection rules for `go-uaa` require to have one approval for the pr. In general we want to encourage the pr workflow and having reviews for transparency and security. If the pr you have, gets a review you should be able to merge. If there is an agreement among all the approvers for the `go-uaa` library to not require reviews this could be changed with static branch protection rule possible via https://github.com/cloudfoundry/community/blob/main/orgs/branchprotection.yml. You will need a pr and agreement between all approvers as mentioned. 

Please let me know what you think about the proposals.

**Comment by beyhan** [comment]
Hi @joefitzgerald,
I wanted  check about what approach would work best for you. We discussed following options until now:

1. If you'd like, you could submit a PR to re-add yourself after we merge https://github.com/cloudfoundry/community/pull/1271. As we discussed, I'm happy to assist with this process, and it would help address the activity check requirements for another year.

2. Alternatively, if you're interested in participating in discussions or other maintenance activities in go-uaa, I'd be glad to remove you from https://github.com/cloudfoundry/community/pull/1271.

Please let me know which option you'd prefer, or if you have any other ideas in mind.

**Pull request merged**
